litemapy/storage.py

```diff
@@ -74,29 +78,32 @@ def __setitem__(self, index, value):
             self.array[end_arr_index] = (self.array[end_arr_index] >> j1 << j1 | (
                         value & self.__mask) >> end_offset) & m
 
-    def __len__(self):
+    def __len__(self) -> int:
         return self.size
 
-    def __iter__(self):
+    def __iter__(self) -> Generator[int, None, None]:
         for i in range(len(self)):
             yield self[i]
 
-    def __reversed__(self):
+    def __reversed__(self) -> 'LitematicaBitArray':
         arr = LitematicaBitArray(self.size, self.nbits)
         for i in range(len(self)):
             arr[i] = self[len(self) - i - 1]
         return arr
 
-    def __contains__(self, value):
+    def __contains__(self, value: int) -> bool:
         for v in self:
             if v == value:
                 return True
         return False
 
 
 class DiscriminatingDictionary(dict):
+    validator: Callable[[Any],tuple[bool, str]]
```

Comment:
`validator` gets both the key and its value as parameters.

```suggestion
    validator: Callable[[Any, Any],tuple[bool, str]]
```